### PR TITLE
test against nettle

### PR DESCRIPTION
**Pull request by jw3126**
Pros: simple, covers a lot of situations
Cons: depends on Nettle

### CODE REVIEW

**Comment by jw3126** [comment]
Okay seems depending on libnettle is painful for CI.

**Comment by jw3126** [comment]
Nettle makes this too fragile for CI. I could save Nettle-precomputed values into a file instead, what do you think @oxinabox?

**Comment by oxinabox** [comment]
Nettle shouldn't make this too fragile for CI.
I'ld like to take a look at nettle and see why it is failing so.

One option is to use Sudo enabled Travis (which is slower), that will fix the linux's.

**Comment by oxinabox** [comment]
At the end of the day though correct is correct.
And a finite battery of precalculated tests should be fine,
since finding hash collisions, even incorrectly implemented MD5 (against correct MD5) by mistake is very unlikely

**Comment by jw3126** [comment]
If CI can be fixed, it is much better then to precalculate files. Files can get big, pollute the git history, are harder to change etc.

**Comment by oxinabox** [comment]
Ok status:

 - Travis is basically not running MacOS jobs but this should be fine.
 - 0.6 Linux is working
 - 0.7 Linux/Windows is broken because Nettle needs to be updated to be 0.7 compatible
 - 64bit 0.6 windows is working
 - 32bit 0.6 windows is not working, but it is not working on master either, see #4

**Comment by jw3126** [comment]
I would say we wait for #4 and then merge this if 32 bit passes. If CI fails annoy us too much in the future, we can replace this PR by lookup files or try to fix Nettle.

File: test/nettle.jl
Comment (oxinabox) [inline test/nettle.jl]: My personal style is to use `in` in forloops,
and reserve  `∈` for operations on sets.

and since the loop later uses `in` this should also for consistancy

File: test/nettle.jl
Comment (oxinabox) [inline test/nettle.jl]: maybe a more descriptive name,
like `md5_ctx` and `nettle_ctx`

or even `md5_value` and `nettle_value`

File: test/nettle.jl
Comment (oxinabox) [inline test/nettle.jl:1]: Can we make it so that this test can be run standalone.
by adding:

```
import MD5
using Base.Test
```

**Review by oxinabox — CHANGES_REQUESTED** [review]
Some minor changes.
also probably good to do a `rebase` so we can see the test results.

Sorry about not getting back to you about them earlier.
I was worrying about the CI failure first, and didn't look closely at the code.

File: test/nettle.jl
Comment (jw3126) [inline test/nettle.jl]: `test_equal_state` runs twice per iteration.

Comment (oxinabox) [reply test/nettle.jl]: true the former can be deleted.

Comment (oxinabox) [reply test/nettle.jl]: true the former can be deleted. I should have caught that.

**Review by oxinabox — APPROVED** [review]

**Comment by oxinabox** [comment]
LGTM (and travis finlly is kind of running again)

**Pull request merged**
